coderd/database/queries/activitybump.sql

```diff
@@ -25,7 +25,7 @@ UPDATE
 SET
 	updated_at = NOW(),
 	deadline = CASE
-		WHEN l.build_max_deadline = '0001-01-01 00:00:00+00'
+		WHEN l.build_max_deadline AT TIME ZONE 'UTC' = '0001-01-01 00:00:00+00'
```

Comment:
Why do we need to remove timezone information here to do the comparison (`AT TIME ZONE` removes it)?

Comparing two times in different timezones is not an issue as long as both are `timestamptz` and represent the same time otherwise:

```
coder=# select '2023-09-14 11:27:08.066479+00'::timestamptz = '2023-09-14 14:27:08.066479+03'::timestamptz;
-[ RECORD 1 ]
?column? | t
```

I don't see being a bug, but it's confusing and since this works there's also no reason to do it. I believe the above `timestamp => timestamptz` change alone was enough.

Reply:
Nice catch. I was mucking around with multiple approaches and stopped once it worked, but it makes sense if only one change was required.